website/static/js/fangorn.js

```diff
@@ -833,6 +836,45 @@ function expandStateLoad(item) {
     reapplyTooltips();
 }
 
+/**
+ * @param tree A Treebeard _item object for the row
+ * @param nodeID Current node._id
+ * @param file window.contextVars.file object
+ */
+function findCurrentFileID(tree, nodeID, file) {
+    var tb = this;
+    if (tb.options.folderIndex !== undefined && tb.options.folderArray !== undefined && tb.options.folderIndex < tb.options.folderArray.length) {
```

Comment:
I don't know, this discussion can become academic; what is important is that the change you make is available to you within other aspects of treebeard. TB itself will always be available and hence tb.options as well but tb.options is the only way to inject anything into Treebeard. So you need a place to add your namespaced variables at such a time that they will be available. For instance if you add it in onload option you will only have it applied to Tb when it gets loaded. I hope this was clear, so the point is you don't have access to tb when you are setting options but your options hooks (like onload or others that are methods) have it.